bokeh/model.py

```diff
@@ -741,7 +727,21 @@ def __exit__(self, type, value, traceback):
 
     def __enter__(self):
         for model in self._to_remove_after:
-            self._doc.add_root(model)
+            try:
+                self._doc.add_root(model)
+            except RuntimeError as e:
+                try:
+                    for r in takewhile(lambda x: x != model, self._to_remove_after):
+                        r.document.remove_root(r)
```

Comment:
What happens when this tries to remove the root that was never added (because it generated the exception, or would have come after that one in the loop)? It might be harmless but I am not certain, so we will need to verify the behavior.